File: Assignment37_2.cpp
//2. Write a program which checks whether 5th & 18th bit is On or OFF. 

#include<iostream>
using namespace std;

class BitOpration
{
    public:
        int iNo;
        int iMask;
        int iResult;

        BitOpration(int A)
        {
            iNo = A;
            iMask = 0X00020010;
            iResult = 0;
        }

        bool ChkBit()
        {
            iResult = iNo & iMask;

            if(iMask == iResult)
            {
                return true;
            }
            else
            {
                return false;
            }

        }

};
int main()
{
    bool bRet = false;
    int iValue = 0;

    cout<<"Enter the Number Here : \n";
    cin>>iValue;
    
    BitOpration obj(iValue);

    bRet = obj.ChkBit();

    if(bRet == true)
    {
        cout<<"5th and 18th Bit is ON....\n";
    }
    else
    {
        cout<<"5th and 18th Bit is OFF....\n";
    }

    return 0;
}

File: Assignment37_2.py
#2. Write a program which checks whether 5th & 18th bit is On or OFF. 

class BitOpration:
    
    iValue = None
    iMask = None
    iResult = None
    
    def __init__(self,iValue):
        self.iValue = iValue
        self.iMask = 0X00020010
        self.iResult = 0
    
    def ChkBit(self):
        self.iResult = self.iValue & self.iMask
        
        if(self.iResult == self.iMask):
            return True
        else:
            return False
        
        

def main():
    
    bRet = None
     
    print("Enter the Number : ")
    iNo = int(input())
    
    Obj = BitOpration(iNo)
    bRet = Obj.ChkBit()
    
    if(bRet == True):
        print("The 5th and 18th Bit are ON...")
    else:
        print("The 5th and 18th Bits are OFF...")
    
    

if __name__=="__main__":
    main()